### DISCUSSION TITLE
Spacy Config for RE task using Transformer

### DISCUSSION THREAD

**Discussion by OpenAIMP** [explosion/spaCy#10301] [Help: Coding & Implementations] [answered]
Hi, can you please provide the command that is used to generate the config files for rel_component? https://github.com/explosion/projects/tree/v3/tutorials/rel_component/configs

**Comment by adrianeboyd** [comment, marked as answer]
These particular configs were probably partially written by hand.

For any component, you should be able to generate a working default config with `spacy init config -p component`, but it probably won't be identical to the configs in the demos. The default configs are usually intended for use with `tok2vec`, so you may need to adjust the config further if you switch to `transformer`.

`spacy init config` supports a lot of the `transformer` adjustments automatically with `-G` for built-in components like `tagger`, but for experimental or custom components, you may need to make more adjustments by hand, also in the `[training]` block.

Currently, we can recommend starting with a `transformer` config for a built-in component (`spacy init config -p tagger -G`) and then swapping the `tagger` out for the custom component config that you want.

You can also compare the two configs in that project to see what the main difference are between `tok2vec` and `transformer` training settings.
